screens/SignUpScreen.js

```diff
@@ -0,0 +1,268 @@
+            <Image
+              style={styles.logoText}
+              source={require("../assets/images/wordmark.png")}
+            />
+            {/* Chose not to use because of padding issue
+          <KitText
+            style={styles.logoText}
+            color={Colors.KIT_BLACK}
+            fontWeight="bold"
+            size={100}
+          >
+            kit
+          </KitText> */}
+          </View>
+          <KitText color={Colors.KIT_DARK_GREY} fontWeight="semibold" size={20}>
+            keep in touch
+          </KitText>
+        </View>
+        <View style={styles.textContainer}>
+          <KitText
+            style={styles.text}
+            color={Colors.KIT_BLACK}
+            size={32}
+            fontWeight="bold"
+          >
+            Hello!
+          </KitText>
+          <KitText
+            style={styles.text}
+            color={Colors.KIT_LIGHT_ORANGE}
+            size={16}
+            fontWeight="semibold"
+          >
+            Create an account to continue.
+          </KitText>
+        </View>
+        <View style={styles.textInputContainer}>
+          <KitTextInput
+            image={require("../assets/images/onboardingname.png")}
+            placeholder="Display name"
+            onBlur={checkDisplayName}
+            onChangeText={displayName => {
+              setDisplayName(displayName);
+              setDisplayNameError(false);
+            }}
+            value={displayName}
+            error={displayNameError}
+            success={displayNameSuccess}
+          />
+          <KitTextInput
+            image={require("../assets/images/onboardingemail.png")}
+            placeholder="Email"
+            onBlur={checkEmail}
+            onChangeText={email => {
+              setEmail(email);
+              setEmailError(false);
+            }}
+            value={email}
+            error={emailError}
+            success={emailSuccess}
+          />
+          <KitTextInput
+            image={require("../assets/images/onboardingpassword.png")}
+            placeholder="Password"
+            onBlur={checkPassword}
+            onChangeText={password => {
+              setPassword(password);
+              setPasswordError(false);
+            }}
+            value={password}
+            error={passwordError}
+            success={passwordSuccess}
+            secureTextEntry
+          />
+          <KitTextInput
+            image={require("../assets/images/onboardingpassword.png")}
+            placeholder="Re-enter password"
+            onBlur={checkReenteredPassword}
+            onChangeText={password => {
+              setReenteredPassword(password);
+              setReenteredPasswordError(false);
+            }}
+            value={reenteredPassword}
+            error={reenteredPasswordError}
+            success={reenteredPasswordSuccess}
+            secureTextEntry
+          />
+          <KitButtonSupreme
+            onPress={async () => {
```

Comment:
This function silently fails if any of the checks fail. I would be nice to have an error message like: `your account could not be created`

Reply:
yeah this is a little confusing/hacky but it actually doesn't fail silently since each of the checks will set an error message. so if you hit register and not all of the checks pass, each check that was not passed will be put in an error state with an error message.

Reply:
nvm something is wrong, will fix